File: client/src/interfaces/Student.ts
import type { Id } from './surrealDb';
import type { User } from './User';

export interface StudentEntity {
  id: Id<'student'>;
  id_card: string;
  user: Id<'user'>;
}

export interface StudentWithUser extends Omit<StudentEntity, 'user'> {
  user: User;
}

File: server/models/student.go
package models

import (
	surrealModels "github.com/surrealdb/surrealdb.go/pkg/models"
)

type StudentEntity struct {
	ID      surrealModels.RecordID `json:"id" swaggertype:"object"`
	Id_card string                 `json:"id_card"`
	User    surrealModels.RecordID `json:"user" swaggertype:"object"`
}

type StudentWithUser struct {
	StudentEntity
	User UserEntity `json:"user"`
}
